lib/skate_web/controllers/detours_controller.ex

```diff
@@ -4,10 +4,25 @@ defmodule SkateWeb.DetoursController do
   alias Skate.OpenRouteServiceAPI
   use SkateWeb, :controller
 
+  alias Skate.Detours.Detours
   alias Skate.Detours.MissedStops
   alias Skate.Detours.RouteSegments
+  alias SkateWeb.AuthManager
   alias Util.Location
 
+  @spec update_snapshot(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def update_snapshot(conn, %{"snapshot" => snapshot} = _params) do
+    uuid = Map.get(snapshot, "uuid")
+
+    %{id: user_id} = AuthManager.Plug.current_resource(conn)
+
+    Detours.update_or_create_detour_for_user(user_id, uuid, %{
+      state: snapshot
+    })
+
+    send_resp(conn, 200, "")
```

Comment:
```suggestion
    send_resp(conn, :ok, "")
```

Reply:
I think we should do something like this though
```suggestion
    if is_nil(uuid) do
      send_resp(conn, :created, new_uuid)
    else
      send_resp(conn, :ok, "")
    end
```

Reply:
And we'd get `new_uuid` from the database upsert

Reply:
Switched to using `json(conn)`